src/en/developer-debugging.md

```diff
@@ -170,21 +170,43 @@ for more information on the automatic retry feature and how to disable this
 behaviour. Note that option `--no-retry` can be used to prevent re-execution of
 failed hooks on the unit.
 
-## Debugging reactive Charms
+## Debugging reactive charms
 
-Debugging a reactive Charm is similar to debugging a regular charm: you use `juju debug-hooks` to get into the hook context and you start the reactive framework by executing the hook file in the `hooks/` directory. That hook will start the reactive framework and [run your handlers based on which flags are set](https://charmsreactive.readthedocs.io/en/latest/dispatch.html). Note that the reactive framework **resets all flags to their original value when a handler crashes.** Changes to flags happen immediately, but they are only persisted at the end of a complete and successful run of the reactive framework. All unpersisted changes are discarded when a hook crashes. This is the case because a Juju hook is transactional: all changes such as open ports are discarded when the hook fails.
+Debugging a reactive charm is similar to debugging a regular charm: 
+use `juju debug-hooks` to get into the hook context and you start the reactive
+framework by executing the hook file in the `hooks/` directory. That hook will 
+start the reactive framework and 
+[run your handlers based on which flags are set](https://charmsreactive.readthedocs.io/en/latest/dispatch.html).
 
-*Tip: If essential hooks such as `install` and `config-changed` are not in the `hooks` directory of your Charm, you forgot to include `layer:basic` in your `layer.yaml` file.*
+Note that the reactive framework **resets all flags to their original value
+when a handler crashes.** Changes to flags happen immediately, but only persist
+at the end of a complete and successful run of the reactive framework. All unpersisted
+changes are discarded when a hook crashes. This is the case because a Juju hook is
+transactional: **all** changes are discarded when the hook fails.
+
+!!! Tip: 
+    If essential hooks such as `install` and `config-changed` are not in the
+    `hooks` directory of your Charm, you forgot to include `layer:basic` in
+    your `layer.yaml` file.*
 
 ### charms.reactive command
 
-charms.reactive has a commandline tool that can help you debugging Charms. You can use it to view, set and clear flags in a debug-hooks session. *Charms built with older versions of the reactive framework might require you to use `state` instead of `flag`.*
+There is a commandline charms.reactive tool that can help with debugging charms.
```

Comment:
My only suggestion is that, as you're touching this doc, it may be worth adding a space to commandline. It's written like this only in one other place in the Juju dev docs (`reference-environment-variables.md`).

Reply:
Fair enough